test/subscribers/publish_zync_event_subscriber_test.rb

```diff
@@ -1,14 +1,67 @@
+# frozen_string_literal: true
+
 require 'test_helper'
 
 class PublishZyncEventSubscriberTest < ActiveSupport::TestCase
-  def setup
-    @subscriber = PublishZyncEventSubscriber.new
+
+  class OIDCApplicationEventTest < ActiveSupport::TestCase
+    attr_reader :event, :subscriber, :publisher
+
+    setup do
+      service = FactoryBot.create(:simple_service, backend_version: 'oauth')
+      application = FactoryBot.create(:simple_cinstance, service: service)
+      @event = Applications::ApplicationCreatedEvent.create(application, nil)
+      @publisher = mock('publisher')
```

Comment:
The `DomainEventsTest` below doesn't need the mock. Are you sure you need it?

Reply:
Well, with this mock I'm doing an extra check that the publisher is actually called (or not called), because the publisher used to be called all the time, but the new logic makes it conditional, so I think we do need to verify whether it's called. And mocking it also helps us control what the publisher returns.